File: simplevlcplayer/src/main/java/com/masterwok/simplevlcplayer/contracts/VlcMediaPlayer.java
package com.masterwok.simplevlcplayer.contracts;

import android.view.SurfaceView;

import org.videolan.libvlc.IVLCVout;
import org.videolan.libvlc.Media;
import org.videolan.libvlc.RendererItem;

import java.io.FileDescriptor;

public interface VlcMediaPlayer
        extends MediaPlayer {

    IVLCVout getVout();

    void attachSurfaces(
            SurfaceView surfaceMedia,
            SurfaceView surfaceSubtitles,
            IVLCVout.OnNewVideoLayoutListener layoutListener
    );

    void detachSurfaces();

    void setRendererItem(RendererItem rendererItem);

    RendererItem getSelectedRendererItem();

    void setAspectRatio(String aspectRatio);

    void setScale(float scale);

    Media.VideoTrack getCurrentVideoTrack();
}

File: simplevlcplayer/src/main/java/com/masterwok/simplevlcplayer/contracts/VlcMediaPlayer.kt
package com.masterwok.simplevlcplayer.contracts

import android.net.Uri
import android.view.SurfaceView

import org.videolan.libvlc.IVLCVout
import org.videolan.libvlc.Media
import org.videolan.libvlc.RendererItem

import java.io.FileDescriptor

interface VlcMediaPlayer : MediaPlayer {

    val vOut: IVLCVout

    val selectedRendererItem: RendererItem?

    val selectedSubtitleUri: Uri?

    val currentVideoTrack: Media.VideoTrack?

    val media: Media?

    fun setMedia(fileDescriptor: FileDescriptor?)

    fun attachSurfaces(
            surfaceMedia: SurfaceView,
            surfaceSubtitles: SurfaceView,
            layoutListener: IVLCVout.OnNewVideoLayoutListener
    )

    fun detachSurfaces()

    fun setRendererItem(rendererItem: RendererItem?)

    fun setVolume(volume: Int)

    fun setAspectRatio(aspectRatio: String?)

    fun setScale(scale: Float)
}